### PR TITLE
runtime-rs: Fix network queues and propagate it to network endpoints

### PR DESCRIPTION
**Pull request by Apokleos**
When creating virtio-net devices in Kata runtime-rs, two multi-queue related fields — `queue_num` and `queue_size` were left missing/default in the device descriptor handed down to the hypervisor. 

Each network endpoint (veth / macvlan / ipvlan / vlan) only populated host_dev_name, virt_iface_name, and guest_mac when building the NetworkPair, while queue_num/queue_size fell back to ..Default::default(). 

As a result, the virtio-net device inside the guest was always created single-queue (queue_num degrading to 1), unable to leverage the host-side multiple vCPUs for parallel RX/TX. 

Consequently, network throughput under high concurrency was limited, and the `network_queues` setting declared in the hypervisor TOML config was never actually consumed by the network path.

This PR aims to address it.

### CODE REVIEW

File: src/runtime-rs/crates/resource/src/network/network_pair.rs [hunk 1]
```diff
@@ -36,6 +36,9 @@ pub struct NetworkPair {
     pub virt_iface: NetworkInterface,
     pub model: Arc<dyn network_model::NetworkModel>,
     pub network_qos: bool,
+    /// Number of virtio queue pairs (each pair = 1 RX + 1 TX).
+    /// Derived from `network_queues` in the hypervisor TOML config.
+    pub queues: usize,
```
[the remark this answers is not in the document]
Comment (Apokleos) [reply src/runtime-rs/crates/resource/src/network/network_pair.rs]: good catch ! fixed

File: src/runtime-rs/crates/resource/src/network/dan.rs [hunk 2]
```diff
@@ -68,6 +68,8 @@ impl DanInner {
         let json_str = fs::read_to_string(&config.dan_conf_path)
             .await
             .context("Read DAN config from file")?;
+        // The `network_queues` comes from hypervisor configurations
+        let default_network_queues = config.network_queues;
```
[the remark this answers is not in the document]
Comment (Apokleos) [reply src/runtime-rs/crates/resource/src/network/dan.rs]: Actually, in the system network_queues will never be zero, as it has been checked when the toml configuration is loaded, if the network_queues is zero and then it will be directly asigned with 1. But I think it might be better to do double-ensure for it.

File: src/runtime-rs/crates/resource/src/network/network_pair.rs [hunk 3]
```diff
@@ -46,6 +49,8 @@ impl NetworkPair {
         model: &str,
         queues: usize,
     ) -> Result<Self> {
+        // Ensure that the queues >= 1
+        let queues = if queues == 0 { 1 } else { queues };
```
Comment (stevenhorsman) [inline src/runtime-rs/crates/resource/src/network/network_pair.rs]: nit: Could we use `let queues = queues.max(1);` as I thinks that is more idiomatic?

File: src/runtime-rs/crates/resource/src/network/endpoint/endpoints_test.rs
Comment (stevenhorsman) [inline src/runtime-rs/crates/resource/src/network/endpoint/endpoints_test.rs:1]: It is worth adding a test with `network_queues: 0` to test the minimum of 1 requirement to avoid any regressions?

File: src/runtime-rs/crates/resource/src/network/dan.rs [hunk 4]
```diff
@@ -80,40 +83,68 @@ impl DanInner {
         let mut entity_list = Vec::with_capacity(config.devices.len());
         for (idx, device) in config.devices.iter().enumerate() {
             let name = format!("eth{idx}");
+            // The `network_queues` is a queue *pair* count.
+            // Keep `queue_num` as a pair count and the hypervisor backend converts pairs into the actual virtqueue count.
+            // A JSON-provided non-zero `queue_num` (also a pair count) with a higher priority always wins.
+            let fallback_queue_num = default_network_queues;
```
Comment (stevenhorsman) [inline src/runtime-rs/crates/resource/src/network/dan.rs]: nit: Do we need this rebind, or could we just use `default_network_queues` below?

File: src/runtime-rs/crates/resource/src/network/dan.rs [hunk 5]
```diff
@@ -80,40 +83,68 @@ impl DanInner {
...
+                    let qnum = if *queue_num == 0 {
+                        fallback_queue_num
+                    } else {
+                        *queue_num
+                    };
+                    let qsize = if *queue_size == 0 {
+                        256
+                    } else {
+                        *queue_size
+                    };
```
Comment (stevenhorsman) [inline src/runtime-rs/crates/resource/src/network/dan.rs]: This is identical to the VHostUser version, so maybe we could extract it? Maybe something like:
```
let qnum = std::num::NonZero::new(*queue_num).map_or(fallback_queue_num, |n| n.get());
let qsize = std::num::NonZero::new(*queue_size).map_or(256, |n| n.get());
```
(or easier to read) might be more idiomatic?

File: src/runtime-rs/crates/resource/src/network/network_pair.rs (hunk 3 above)
Comment (stevenhorsman) [reply src/runtime-rs/crates/resource/src/network/network_pair.rs]: It looks like `.max(1)` is used in the rest of the commits, so that's stronger evidence to suggest swapping.

**Review by stevenhorsman — APPROVED** [review]
Some nits, but the general code seems okay, once the CI is happy

File: src/runtime-rs/crates/resource/src/network/endpoint/endpoints_test.rs
Comment (Apokleos) [reply src/runtime-rs/crates/resource/src/network/endpoint/endpoints_test.rs:1]: fixed

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Apokleos!
